### PR TITLE
Add DDG Index with Loading, Error, and temporary header and done views

### PR DESCRIPTION
**Pull request by everett980**
Implements Loading and Error views, as well as a temporary header that technically works to provide query parameters and a temporary done view that just says loaded

### CODE REVIEW

File: packages/jaeger-ui/src/model/ddg/extractQuery.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const returnVal = arg[0];
+    console.warn(`Found multiple query parameters: "${arg}", using "${returnVal}"`); // eslint-disable-line no-console
+    return returnVal;
+  }
+  return arg;
+}
+
+type TExtracted = { service?: string; operation?: string; start?: number; end?: number };
+
+export default function extractQuery(search: string = window.location.search): TExtracted {
```
Comment (everett980) [inline packages/jaeger-ui/src/model/ddg/extractQuery.tsx]: no default value here

Comment (everett980) [inline packages/jaeger-ui/src/model/ddg/extractQuery.tsx]: this file can live in `components/DeepDependencyGraph/url.tsx`

File: packages/jaeger-ui/src/components/DeepDependencyGraph/url.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    const returnVal = arg[0];
+    console.warn(`Found multiple query parameters: "${arg}", using "${returnVal}"`); // eslint-disable-line no-console
+    return returnVal;
+  }
+  return arg;
+}
+
+type TUrlState = { service?: string; operation?: string; start?: number; end?: number };
+
+export function getUrlState(search: string): TUrlState {
```
Comment (everett980) [inline packages/jaeger-ui/src/components/DeepDependencyGraph/url.tsx]: this is memoized in [this pr](https://github.com/everett980/jaeger-ui/pull/5)

File: packages/jaeger-ui/src/components/DeepDependencyGraph/url.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+function firstParam(arg: string | string[]): string {
+  if (Array.isArray(arg)) {
+    const returnVal = arg[0];
+    console.warn(`Found multiple query parameters: "${arg}", using "${returnVal}"`); // eslint-disable-line no-console
+    return returnVal;
+  }
+  return arg;
+}
+
+type TUrlState = { service?: string; operation?: string; start?: number; end?: number };
```
Comment (everett980) [inline packages/jaeger-ui/src/components/DeepDependencyGraph/url.tsx]: this is moved to `model/ddg/types` in [this pr](https://github.com/everett980/jaeger-ui/pull/5)

File: packages/jaeger-ui/src/components/DeepDependencyGraph/url.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+export function getUrlState(search: string): TUrlState {
+  const { service, operation, start, end } = queryString.parse(search);
+  const returnVal: TUrlState = {};
```
Comment (everett980) [inline packages/jaeger-ui/src/components/DeepDependencyGraph/url.tsx]: this is shortened to rv in [this pr](https://github.com/everett980/jaeger-ui/pull/5)

**Review by tiffon — APPROVED** [review]
Looks good.

**Pull request merged**